### PR TITLE
Route associating

### PR DESCRIPTION
**Pull request by msecret**
To associate a route has a single action, not with creation first.

### CODE REVIEW

**Comment by msecret** [comment]
Requires styling here: https://github.com/18F/cg-style/pull/153

File: static_src/components/route.jsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    );
+  }
+
+  get actions() {
+    const actions = [];
+    const route = this.props.route;
+    if (!route) return actions;
+
+    if (!route.app_guid) actions.push(this.deleteAction);
+    if (route.app_guid) actions.push(this.editAction);
```
Comment (jcscottiii) [inline static_src/components/route.jsx]: use `else` / `else if` instead

File: static_src/components/route.jsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    actions.push(this.bindAction(!!route.app_guid));
+
+    return actions;
+  }
+
+  get confirmationMsg() {
+    return (this.props.route.app_guid) ? 'Unbind this route from this app?' :
```
Comment (jcscottiii) [inline static_src/components/route.jsx]: here, line 117, line 118 and line 119 all do this logic for checking the existence of `app_guid`.  you should create a helper function that does something like `isRouteBoundToApp(route)

File: static_src/components/route_form.jsx [hunk 3]
```diff
@@ -93,31 +94,29 @@ export default class RouteForm extends React.Component {
...
-    );
-  }
-
-  // If there is a delete handler, this form is for editing a route
-  // Otherwise, it is for a new route
-  // The submit action label should be adjusted accordingly
   get submitActionText() {
-    if (this.props.deleteHandler) return 'Apply';
-    return 'Add route';
+    if (this.props.route.guid) return 'OK';
```
Comment (jcscottiii) [inline static_src/components/route_form.jsx]: also here could use the helper.

**Comment by jcscottiii** [comment]
also for some of the codeclimate issues that are in tests files, you might want to add an exclude. we do it similarly here: https://github.com/opencontrol/compliance-masonry/blob/master/.codeclimate.yml#L10

**Pull request merged**